TransportationSplitter.py

```diff
@@ -190,31 +195,26 @@ def are_different_coords(coords1, coords2):
 def round_point(point, precision=POINT_PRECISION):
     return Point(round(point.x, precision), round(point.y, precision))
 
-def split_line(original_line_geometry: LineString, split_points: [SplitPoint] ) -> [SplitSegment]:
+def split_line(original_line_geometry: LineString, split_points: list[SplitPoint] ) -> list[SplitSegment]:
     """Split the LineString into segments at the given points"""
 
     # Special case to avoid processing when there are only start/end split points
     if len(split_points) == 2 and split_points[0].lr == 0 and split_points[1].lr == 1:
         return [SplitSegment(0, original_line_geometry, split_points[0], split_points[1])]
 
-    original_coordinates = list(original_line_geometry.coords)
     split_segments = []
-    last_coordinate_index = 0
-    last_coordinate_dist = 0.0
-    last_segment_coordinates = [split_points[0].geometry.coords[0]] # initialize first split segment with first split point
-
-    remaining_line_geometry = original_line_geometry
-
-    for i in range(len(split_points) - 1):
-        next_split_geometry = split_points[i + 1].geometry
-        snapped_geometry = ops.snap(remaining_line_geometry, next_split_geometry, tolerance=0.0000001)
-        split_line = ops.split(snapped_geometry, next_split_geometry)
-        split_segment_geometry = remove_consecutive_dupe_coords(LineString(split_line.geoms[0]))
-        if len(split_line.geoms) > 1:
-            # Only use remainder of line for snapping to minimize chance of snapping to the wrong location when segment is self-intersecting
-            remaining_line_geometry = split_line.geoms[1]
-        split_segments.append(SplitSegment(i, split_segment_geometry, split_points[i], split_points[i + 1]))
-
+    i=0
+    for split_point_start, split_point_end in zip(split_points[:-1], split_points[1:]):
```

Comment:
Maybe just use a index/range iterator for this? You're already tracking an index with `i`